### DISCUSSION TITLE
Distributed barrier in Jax

### DISCUSSION THREAD

**Discussion by neel04** [jax-ml/jax#22299] [Q&A] [answered]
I'm doing some dataset preprocessing and hence want to sync up all the jax processes on every node/host. However, that [experimental capability](https://jax.readthedocs.io/en/latest/_autosummary/jax.experimental.host_callback.barrier_wait.html) has been phased out.

How can I use a barrier?

And if that's not possible, how can I increase the timeout of the `DEADLINE EXCEEDED: barrier timed out` barrier? Here is the [error log](https://rentry.co/yry6scru) incase someone is curious.

**Comment by yashk2810** [comment, marked as answer]
Try using `multihost_utils.sync_global_devices`?

**Reply by neel04** [reply]
That's it - thank you so much!
